tapdb/assets_store.go

```diff
@@ -722,15 +749,33 @@ func dbAssetsToChainAssets(dbAssets []ConfirmedAsset,
 			AnchorMerkleRoot:       sprout.AnchorMerkleRoot,
 			AnchorTapscriptSibling: sprout.AnchorTapscriptSibling,
 		}
+
+		// We only set the lease info if the lease is actually still
+		// valid and hasn't expired.
+		owner := sprout.AnchorLeaseOwner
+		expiry := sprout.AnchorLeaseExpiry
+		if len(owner) > 0 && expiry.Valid &&
+			expiry.Time.UTC().After(a.clock.Now().UTC()) {
```

Comment:
nit: the if condition is getting complicated here.

I would consider evaluating a bool variable before the if clause:
```
var (
    hasOwner = len(owner) > 0
    isExpired = expiry.Valid && expiry.Time.UTC().After(a.clock.Now().UTC())
)

if hasOwner && isExpired {
```

Or maybe not, it's a style.